landoapi/transplants.py

```diff
@@ -141,6 +138,60 @@ class TransplantAssessmentState:
     # State required for assessing landing requests.
     landing_assessment: Optional[LandingAssessmentState] = None
 
+    @classmethod
+    def from_assessment(
+        cls,
+        phab: PhabricatorClient,
+        stack_data: RevisionData,
+        stack: RevisionStack,
+        landable_repos: dict[str, Repo],
+        supported_repos: dict[str, Repo],
+        reviewers: dict,
+        users: dict,
+        projects: dict,
+        data_policy_review_phid: str,
+        relman_group_phid: str,
+        secure_project_phid: str,
+        testing_tag_project_phids: list[str],
+        testing_policy_phid: str,
+        landing_assessment: Optional[LandingAssessmentState] = None,
+    ) -> StackAssessmentState:
+        """Build a `StackAssessmentState` from passed arguments.
+
+        Build any fields that are shared between checks but are derived from
+        existing fields.
+        """
+        # Create a copy of the stack where we will remove revisions that are blocked from
+        # landing, leaving a graph where each path is landable.
```

Comment:
```suggestion
        # Create a copy of the stack so that revisions that are blocked from landing
        # can be removed from it when parsing `StackAssessmentState`.
```

Reply:
Updated this but kept the note about paths in this stack being landable.